pkg/v3/postprocessors/ineligible_test.go

```diff
@@ -0,0 +1,114 @@
+package postprocessors
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"log"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/smartcontractkit/ocr2keepers/pkg/v3/types"
+)
+
+func TestNewIneligiblePostProcessor(t *testing.T) {
+	for _, tc := range []struct {
+		name         string
+		stateUpdater types.UpkeepStateUpdater
+		results      []types.CheckResult
+		wantLog      string
+		expectsErr   bool
+		wantErr      error
+	}{
+		{
+			name: "upkeep state is updated as per one check result",
+			stateUpdater: &mockStateUpdater{
+				SetUpkeepStateFn: func(ctx context.Context, result types.CheckResult, state types.UpkeepState) error {
+					return nil
+				},
+			},
+			results: []types.CheckResult{
+				{
+					PipelineExecutionState: 0,
+					Eligible:               false,
+				},
+			},
+			wantLog: "post-processing 1 results, 0 ineligible",
+		},
+		{
+			name: "upkeep state is updated as per multiple check results",
+			stateUpdater: &mockStateUpdater{
+				SetUpkeepStateFn: func(ctx context.Context, result types.CheckResult, state types.UpkeepState) error {
+					return nil
+				},
+			},
+			results: []types.CheckResult{
+				{
+					PipelineExecutionState: 0,
+					Eligible:               false,
+				},
+				{
+					PipelineExecutionState: 0,
+					Eligible:               false,
+				},
+				{
+					PipelineExecutionState: 0,
+					Eligible:               false,
+				},
+			},
+			wantLog: "post-processing 3 results, 0 ineligible",
+		},
+		{
+			name: "upkeep state errors",
+			stateUpdater: &mockStateUpdater{
+				SetUpkeepStateFn: func(ctx context.Context, result types.CheckResult, state types.UpkeepState) error {
+					return errors.New("upkeep state boom")
+				},
+			},
+			results: []types.CheckResult{
+				{
+					PipelineExecutionState: 0,
+					Eligible:               false,
+				},
+				{
+					PipelineExecutionState: 0,
+					Eligible:               true,
+				},
+				{
+					PipelineExecutionState: 0,
+					Eligible:               false,
+				},
+			},
+			expectsErr: true,
+			wantErr: errors.New(`upkeep state boom
+upkeep state boom`),
```

Comment:
💥

Reply:
This is a multi error check, so the new line is needed

Reply:
i was just pointing out the boom :p